### PR TITLE
feat: add flags for default execution version

### PR DESCRIPTION
**Pull request by strangelookingnerd**
Another go at https://github.com/camunda/camunda-modeler/pull/3528 with added tests and implementations for all editors.
Closes https://github.com/camunda/camunda-modeler/issues/3515 / https://github.com/camunda/camunda-modeler/issues/3696

Ping @nikku FYI

I tried to add the original remarks of @barmac and additionally added configuration for both, cloud and platform versions.
Please let me know what you think.

### CODE REVIEW

File: client/src/util/Flags.js [hunk 1]
```diff
@@ -46,3 +46,5 @@ export const SENTRY_DSN = 'sentry-dsn';
 export const MIXPANEL_TOKEN = 'mixpanel-token';
 export const MIXPANEL_STAGE = 'mixpanel-stage';
 export const DISPLAY_VERSION = 'display-version';
+export const CLOUD_ENGINE_VERSION = 'cloud-engine-version';
+export const PLATFORM_ENGINE_VERSION = 'platform-engine-version';
```
Comment (barmac) [inline client/src/util/Flags.js:50]: Let's use product names in flags as platform and cloud are our internal legacy names:
```suggestion
export const CLOUD_ENGINE_VERSION = 'c8-engine-version';
export const PLATFORM_ENGINE_VERSION = 'c7-engine-version';
```

**Review by barmac — CHANGES_REQUESTED** [review]
First and foremost, thanks for this contribution. It already looks great and I am super happy about the quality!

Still, in order to make it production-ready, we should make sure that the engine version set by the user is available in the application (https://github.com/camunda/camunda-modeler/blob/develop/client/src/util/Engines.js). Otherwise, an invalid flag could make the editor unusable. So, if the value does not appear on the list, it should be replaced by the latest.

Please let me know whether you want to implement this or should I pick it up from there.

**Comment by strangelookingnerd** [comment]
@barmac Good point - right now an error is thrown but some fallback logic would be better.

To be fair I must admit that JavaScript is not my natural habitat and a solution would likely not be up to your standards.
So feel free to pick it up from here, happy to learn something new 😉

**Comment by barmac** [comment]
I pushed changes which implement https://github.com/camunda/camunda-modeler/issues/3515

Regarding _default execution platform version if not present on the diagram_, I will look into this on Monday.

The error is now swallowed. Probably we could log it easily to developer tools, but having access to application log is a new story.

**Comment by barmac** [comment]
I am closing this PR in favor of https://github.com/camunda/camunda-modeler/pull/3720. Let me know if this is enough for your use case and feel free to contribute to the discussion on https://github.com/camunda/camunda-modeler/pull/3697#issuecomment-1628560528

**Comment by strangelookingnerd** [comment]
@barmac 

> I gave it a thought in the morning and I am not sure whether we really want to make app behave as if the execution platform version was set to a specific value. That is against the current approach which reflects what is set in the XML. If we implement the requested feature, users won't have incentive to set the platform version on preexisting diagrams.

For us it has shown that users often do not even notice that they have not set the platform version yet - this is why I intended the change in the first place.
The idea would not be to change the current default behavior, of course. But that does not rule out the possibility to make it configurable via flags, does it? I still would like to see this feature added to at least give the chance to have a pre-selection since it still remains the users _active_ choice to have this set or not. 

I'm also thinking one should not "override" what ever is set on an existing the diagram, however what if nothing is set on the diagram - at least for me this is still a use case.

So by the end of the day it should be `Version set in Diagram > Version set via Flags > Unset`. 

To explain further on this:
We bundle the modeler together with some (custom) plugins / linter rules and some pre-defined flags alongside some BPMN templates. 
These templates currently do not have a "hard-coded" engine version. We could obviously change that, but since we want to use the same templates (=files) for multiple versions of these bundles it is not really desirable either.
These bundles are then given to users that work with the modeler - and especially the templates - to create processes that run against a specific version of the platform.
If we could pre-define this platform version is the bundle right away, there would be no need to adapt the templates for every delivery as well as giving the user the right settings (=flags) for the environment the bundle was intended for.

> There is however no obstacle to configure the default version for new diagrams (#2491). This is the only feature I'd merge right now, and leave the other for further discussion.

That's great, thank you already.

> As a side note, maybe the execution platform version should not be bound to a diagram but rather to a specific Desktop Modeler instance (or project). As a user, I'd expect linting to follow my cluster's configuration across all of the diagrams. Still, this a change in the app which should be discussed before we start implementing.
> 
> We need to make sure what user's problem we want to solve, and tailor the solution to exactly that problem.

I agree 100% - IMHO - the diagrams should not care for the engine version. From a technical perspective they need to care for the modeler version in order to be "compatible" I guess. But making the execution platform bound to the modeler would be what I'm looking for.

**Pull request closed without merging**